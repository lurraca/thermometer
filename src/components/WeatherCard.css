.weather-card {
    border: 1px solid rgb(185,185,185);
    display: inline-block;
    width: 31%;
    margin: 0 10px 0 0;
}

.weather-card-content {
    padding: 0 0 0 10px;
}