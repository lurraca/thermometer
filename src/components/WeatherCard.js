import React, { Component } from 'react';

import './WeatherCard.css';

class WeatherCard extends Component {
  render() {
    return(
      <div className="weather-card">
        <div className="weather-card-content">
          <p>{this.props.data.avgTemp}Â° C</p>
          <p>{this.props.data.weatherDesc}</p>
          <p>Humidity: {this.props.data.humidity}</p>
        </div>
      </div>
    )
  }
}

export default WeatherCard;