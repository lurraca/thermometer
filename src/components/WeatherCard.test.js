import React from 'react';
import { mount, shallow } from 'enzyme';

import WeatherCard from './WeatherCard';

it('renders correctly WeatherCard ', () => {
  const forecast = {
    "avgTemp": "2",
    "weatherDesc": "rainy",
    "humidity": 100
  };

  const renderedComponent = shallow(
    <WeatherCard data={forecast} />
  );

  expect(renderedComponent).toMatchSnapshot();
});