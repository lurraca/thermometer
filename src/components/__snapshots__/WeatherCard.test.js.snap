// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly WeatherCard  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="weather-card"
>
    <div
        className="weather-card-content"
    >
        <p>
            2
            째 C
        </p>
        <p>
            rainy
        </p>
        <p>
            Humidity: 
            100
        </p>
    </div>
</div>,
  "nodes": Array [
    <div
      className="weather-card"
>
      <div
            className="weather-card-content"
      >
            <p>
                  2
                  째 C
            </p>
            <p>
                  rainy
            </p>
            <p>
                  Humidity: 
                  100
            </p>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <WeatherCard
        data={
                Object {
                        "avgTemp": "2",
                        "humidity": 100,
                        "weatherDesc": "rainy",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WeatherCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "avgTemp": "2",
            "humidity": 100,
            "weatherDesc": "rainy",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="weather-card"
>
          <div
                    className="weather-card-content"
          >
                    <p>
                              2
                              째 C
                    </p>
                    <p>
                              rainy
                    </p>
                    <p>
                              Humidity: 
                              100
                    </p>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="weather-card"
>
          <div
                    className="weather-card-content"
          >
                    <p>
                              2
                              째 C
                    </p>
                    <p>
                              rainy
                    </p>
                    <p>
                              Humidity: 
                              100
                    </p>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <WeatherCard
    data={
        Object {
            "avgTemp": "2",
            "humidity": 100,
            "weatherDesc": "rainy",
          }
    }
/>,
}
`;
